pkg/transform/kubernetes/kubernetes.go

```diff
@@ -321,6 +321,11 @@ func populateEnvFrom(app *spec.App) error {
 				envs = append(envs, app.PodSpec.Containers[ci].Env...)
 				app.PodSpec.Containers[ci].Env = envs
 
+				// this should be done when population is done
+				// TODO: when you add support for envFrom secret
+				// TODO: need to move this after secrets are also handled
+				app.PodSpec.Containers[ci].EnvFrom = nil
```

Comment:
Should this not be in the `for ci, c := range app.Containers` loop. Here, it is run for every configMap.
Just after the `c.EnvFrom` loop -
```golang
			}
		}
		if c.EnvFrom != nil {
			app.PodSpec.Containers[ci].EnvFrom = nil
		}
	}
	return nil
}
```